### DISCUSSION TITLE
How to use Marshal and Unmarshal components?

### DISCUSSION THREAD

**Discussion by dirooster** [apache/camel-karavan#1454] [Q&A] [answered]
### Ask a question

Hi!
I have a question about Camel Karavan. When I use unmarshal/marshal components I should to fill - Context Path field (Package name where your JAXB classes are located). But I dont understand how to add using UI new package with classes, do you have any samples?

### Variant

Web Application

### Container Management (if applicable)

Kubernetes

### Operating System (if applicable)

Windows

### Version

4.8.0

**Comment by dirooster** [comment]
@mgubaidullin, hi 
Sorry to interrupt, could you help me with some additional information?

**Comment by mgubaidullin** [comment, marked as answer]
1. You could create or upload Java files. Package name should be empty or aligned with `camel.jbang.gav` property in `application.properties` 
2. Create custom jar with classes (outside of Karavan) and add is as a dependency in `application.properties`

**Reply by dirooster** [reply]
Thank you for answer
And another couple questions, in this case, which path should be specified to the package, with classes?

Also if I created Java class, but I need to add dependencies that are not in project, how can I upgrade maven file with dependencies or I should also upload custom jar?

**Comment by dirooster** [comment]
@mgubaidullin, sorry to bother you, could you give more information?
